File: engage-4 (Angular and Spring)/engage/src/app/models/project/project-member.ts
import {Project} from './project';
import {User} from '../user/user.model';

export class ProjectMember {
  id: number;
  project: Project;
  user: User;
  owner: boolean;
  role: string;

  constructor(project: Project, user: User, owner: boolean, role: string) {
    this.project = project;
    this.user = user;
    this.owner = owner;
    this.role = role;
  }

}

File: engage-4 (Angular and Spring)/engageserver/src/main/java/app/models/project/ProjectMember.java
package app.models.project;

import app.models.user.User;

import javax.persistence.*;

@Entity
public class ProjectMember {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;
    @ManyToOne
    private Project project;
    @OneToOne
    private User user;
    private boolean owner;
    private String role;

    public ProjectMember() {

    }

    public ProjectMember(Project project, User user, boolean owner, String role) {
        this.project = project;
        this.user = user;
        this.owner = owner;
        this.role = role;
    }

    public long getId() {
        return id;
    }

    public Project getProject() {
        return project;
    }

    public User getUser() {
        return user;
    }

    public boolean isOwner() {
        return owner;
    }

    public String getRole() {
        return role;
    }

    @Override
    public String toString() {
        return "ProjectMember{" +
                "id=" + id +
                ", project=" + project +
                ", owner=" + owner +
                ", role=" + role +
                '}';
    }
}
